### PR TITLE
T dh 551

### PR DESCRIPTION
**Pull request by Fitchburgh**

### CODE REVIEW

File: src/aura/ACSRecordPreviewer/ACSRecordPreviewerController.js [hunk 1]
```diff
@@ -1,117 +1,171 @@
...
+                        console.log(accountList[i].a)
+                        component.set("v.accountToRelate", accountList[i].a);
+                        return;
+                    }
+                }
+            }
+        });
+    },
+
+    handleComponentContactList: function(component) {
```
Comment (byarsawich) [inline src/aura/ACSRecordPreviewer/ACSRecordPreviewerController.js]: What does this actually do?

File: src/aura/ACSRecordPreviewer/ACSRecordPreviewerController.js [hunk 2]
```diff
@@ -1,117 +1,171 @@
...
+        helper.accountInfoHelper(component, accounts);
+    },
+
+    afterScriptsLoaded: function(component, event, helper) {
+        jQuery("document").ready(function(){
+            console.log('loading jquery lib');
+        });
+    },
+
+    initDrag: function(component, event, helper) {
```
Comment (byarsawich) [inline src/aura/ACSRecordPreviewer/ACSRecordPreviewerController.js]: :jail:

File: src/classes/ACSRecordPreviewerController.cls [hunk 3]
```diff
@@ -1,107 +1,108 @@
...
+            system.debug(rpw);
+            rpwList.add(rpw);
+        }
+        system.debug(rpwList);
+        return rpwList;
+    }
+
+    @AuraEnabled
+    public static List<String> getRoles()
+    {
```
Comment (byarsawich) [inline src/classes/ACSRecordPreviewerController.cls]: style guide

File: src/classes/ACSRecordPreviewerController.cls [hunk 4]
```diff
@@ -1,107 +1,108 @@
...
+ * Created by bfitch on 7/9/2018.
+ */
+
+public with sharing class ACSRecordPreviewerController {
+    @AuraEnabled
+    public static List<RecordPreviewerWrapper> accountDataByFS(List<Account> accList) {
+        List<RecordPreviewerWrapper> rpwList = new List<RecordPreviewerWrapper>();
+        system.debug(accList);
+
+        for(Account acc : accList) {
```
Comment (byarsawich) [inline src/classes/ACSRecordPreviewerController.cls]: 3 SOQL statements inside a loop is not good

File: src/aura/ACSRecordPreviewer/ACSRecordPreviewerController.js (hunk 1 above)
Comment (Fitchburgh) [reply src/aura/ACSRecordPreviewer/ACSRecordPreviewerController.js]: Nothing, it's gone.

File: src/classes/ACSRecordPreviewerController.cls (hunk 3 above)
Comment (Fitchburgh) [reply src/classes/ACSRecordPreviewerController.cls]: fixed

File: src/classes/ACSRecordPreviewerController.cls [hunk 5]
```diff
@@ -1,107 +1,108 @@
...
+    @AuraEnabled
+    public static AccountContactRelation assignRole(Id con, String role, Id acc) {
+        system.debug(con);
+        system.debug(role);
+        system.debug(acc);
+
+        AccountContactRelation acr = new AccountContactRelation();
+        AccountContactRelation old = new AccountContactRelation();
+        try {
+            old =
```
[the remark this answers is not in the document]
Comment (Fitchburgh) [reply src/classes/ACSRecordPreviewerController.cls]: This cannot return more than one record.

File: src/classes/ACSRecordPreviewerController.cls [hunk 6]
```diff
@@ -22,14 +31,13 @@ public with sharing class ACSRecordPreviewerController {
 
             system.debug(acc);
 
-            // search AccountFieldSet__mdt to find the name
-            AccountFieldSet__mdt afs =
-                    [SELECT Field_Set_Dev_Name__c, Record_Type_Dev_Name__c
-                    FROM AccountFieldSet__mdt
-                    WHERE Record_Type_Dev_Name__c = :rType];
-
+            Map<String, sObject> fieldMap = ArrayUtils.getMaptoField(afsList, 'Record_Type_Dev_Name__c');
```
Comment (byarsawich) [inline src/classes/ACSRecordPreviewerController.cls]: Move this out of the loop, old needs to be done once since it's a static list

**Pull request merged**